GDJS/Runtime/variablescontainer.ts

```diff
@@ -220,15 +220,29 @@ namespace gdjs {
     getStructureNetworkSyncData(
       variable: gdjs.Variable
     ): VariableNetworkSyncData[] | undefined {
-      const variableChildren =
-        variable.getType() === 'structure' ? variable.getAllChildren() : null;
+      if (variable.getType() === 'array') {
+        return variable.getAllChildrenArray().map((childVariable) => {
+          return {
+            name: '',
+            value: childVariable.getValue(),
+            type: childVariable.getType(),
+            children: this.getStructureNetworkSyncData(childVariable),
+          };
+        });
+      }
+
+      if (variable.getType() !== 'structure') {
+        return undefined;
+      }
+
+      const variableChildren = variable.getAllChildren();
 
       const childrenSyncData = variableChildren
         ? Object.entries(variableChildren).map(
             ([childVariableName, childVariable]) => {
               return {
                 name: childVariableName,
-                value: childVariable.getAsNumberOrString(),
+                value: childVariable.getValue(),
```

Comment:
I would check the type first. If it's not a primitive, don't even call getValue. Same for array. It's possible getValue will do internal stuff that would change the variable type or make things worse for arrays/structures as the runtime could think you're trying to use it as a primitive